src/net.cpp

```diff
@@ -962,6 +962,16 @@ static bool CompareNodeTXTime(const NodeEvictionCandidate &a, const NodeEviction
     return a.nTimeConnected > b.nTimeConnected;
 }
 
+
+//! Sort an array by the specified comparator, then remove the last K elements.
+template<typename T>
+static void removeLastKElements(std::vector<T> &elements, bool(*comparator)(const T&, const T&), size_t k)
```

Comment:
Maybe don't require comparator to be function pointer. It'd be nice to allow comparator to be a lambda or function class:

```c++
template<typename T, typename Compare>
static void removeLastKElements(std::vector<T> &elements, Compare comp, size_t k)
{
    std::sort(elements.begin(), elements.end(), comp);
    ...
}
```

Reply:
That's a good point, should have pulled that from your example.  Old habits die hard.